executor/builder.go

```diff
@@ -4962,8 +4963,14 @@ func (b *executorBuilder) buildWindow(v *plannercore.PhysicalWindow) exec.Execut
 				exec.orderByCols = orderByCols
 				exec.expectedCmpResult = cmpResult
 				exec.isRangeFrame = true
-				exec.start.InitCompareCols(b.ctx, exec.orderByCols)
-				exec.end.InitCompareCols(b.ctx, exec.orderByCols)
+				err = exec.start.UpdateCompareCols(b.ctx, exec.orderByCols)
+				if err != nil {
+					return nil
+				}
+				err = exec.end.UpdateCompareCols(b.ctx, exec.orderByCols)
```

Comment:
Please double comfirm that since `UpdateCompareCols` is already called during `BuildWindowFunctionFrameBound`, is it necessary to call it again here.

Reply:
> Please double comfirm that since `UpdateCompareCols` is already called during `BuildWindowFunctionFrameBound`, is it necessary to call it again here.

`UpdateCompareCols` is already called in `buildWindowFunctionFrameBound`, however, we need to call it again as the meta data in orderByCols may be changed.